### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by MakArs**
A bit of refactoring and covering with unit tests

### CODE REVIEW

**Review by gabbacode — COMMENTED** [review]
probably better to use 
using var ...
?

File: ReportService/ReportTask/ReportTask.cs [hunk 1]
```diff
@@ -21,67 +21,56 @@ public partial class ReportTask : IReportTask
...
-        public Dictionary<string, object> Parameters { get; set; } //todo: change to class?
-        public List<ParameterInfo> ParameterInfos { get; set; }
-        public List<TaskDependency> DependsOn { get; set; }
+        public List<IOperation> Operations { get; set; } = new List<IOperation>();
+        public Dictionary<string, object> Parameters { get; set; } = new Dictionary<string, object>();//todo: change to class?
+        public List<ParameterInfo> ParameterInfos { get; set; } = new List<ParameterInfo>();
+        public List<TaskDependency> DependsOn { get; set; } = new List<TaskDependency>();
 
-        private readonly IMonik monik;
-        private readonly ILifetimeScope autofac;
```
Comment (gabbacode) [inline ReportService/ReportTask/ReportTask.cs:30]: is scope or container more precise naming?

**Review by gabbacode — APPROVED** [review]

**Review by gabbacode — APPROVED** [review]

**Review by gabbacode — DISMISSED** [review]

File: ReportService/ReportTask/ReportTask.cs (hunk 1 above)
Comment (MakArs) [reply ReportService/ReportTask/ReportTask.cs:30]: Makes sense.

**Review by gabbacode — DISMISSED** [review]

File: ReportService/Operations/DataExporters/Dependencies/B2BDbStructureChecker.cs [hunk 2]
```diff
@@ -28,19 +26,15 @@ public class B2BDbStructureChecker : DBStructureChecker
...
-        #region overrides
 
         public override async Task<bool> CheckIfDbStructureExists(DbConnection connection, IReportTaskRunContext taskContext)
         {
-            var result = await connection.QueryFirstOrDefaultAsync<int>(new CommandDefinition(DbStructureCheckQuery,
-                     new { taskId = taskContext.TaskId }, commandTimeout: DbTimeOut,
-                     cancellationToken: taskContext.CancelSource.Token));
+            Guard.Against.Null(taskContext, nameof(taskContext));
+
+            int result = await connection.QueryFirstOrDefaultAsync<int>(new CommandDefinition(DbStructureCheckQuery, new { taskId = taskContext.TaskId },
```
Comment (gabbacode) [inline ReportService/Operations/DataExporters/Dependencies/B2BDbStructureChecker.cs:34]: is wide strings better than short?

**Review by gabbacode — APPROVED** [review]

File: ReportService/Operations/DataExporters/Dependencies/B2BDbStructureChecker.cs (hunk 2 above)
Comment (MakArs) [reply ReportService/Operations/DataExporters/Dependencies/B2BDbStructureChecker.cs:34]: For me now it really improves readability in most of cases (easier to see where one line/declaration ends and next starts). But it is more matter of taste and size of monitor

Comment (gabbacode) [reply ReportService/Operations/DataExporters/Dependencies/B2BDbStructureChecker.cs:34]: 1. eyes do 10 times less work if each parameter starts in one column
2. it's hard to find somth in one long string
3. don't need to scroll left to right
4. use ident so you never ever lost your string ending

ex:
Method(
    param1,
    param2,
    anotherOne);

nextString

**Pull request merged**